File: Programs/Part-A/Program-3/Java/app/src/main/java/com/demo/code/LogInActivity.java
package com.demo.code;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class LogInActivity extends AppCompatActivity {

    private String userName,password;

    private EditText usrNameId;
    private EditText pwdId;
    private Button loginId;

    private int attemptCounter = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_log_in);
        findViewsForTheScreen();
        getDataFromPreviousScreen();
        setOnClickListener();
    }

    private void setOnClickListener() {


        loginId.setOnClickListener(v -> {
            attemptCounter++;
            String loginUserName = usrNameId.getText().toString();
            String loginPassword = pwdId.getText().toString();
            if(attemptLogin(loginUserName,loginPassword)){
                //Navigate to success activity
                Intent intent = new Intent(LogInActivity.this, SuccessActivity.class);
                startActivity(intent);
            }else{
                if(attemptCounter>=2){
                    // Disable the Login Button
                    loginId.setEnabled(false);
                }
            }
        });
    }

    private boolean attemptLogin(String loginUserName, String loginPassword) {

        if(userName.equals(loginUserName) && password.equals(loginPassword)){
            return true;
        }else{
            Toast.makeText(LogInActivity.this,"Login Failed",Toast.LENGTH_SHORT).show();
            return false;
        }

    }

    private void getDataFromPreviousScreen() {
        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            userName = bundle.getString("userName");
            password = bundle.getString("password");
        }
    }

    /**
     * Find the objects for the screen
     */
    private void findViewsForTheScreen() {
        usrNameId = findViewById(R.id.usrNameId);
        pwdId = findViewById(R.id.pwdId);
        loginId = findViewById(R.id.loginId);
    }

}

File: Programs/Part-A/Program-3/Kotlin/app/src/main/java/com/demo/code/LogInActivity.kt
package com.demo.code

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.demo.code.databinding.ActivityLogInBinding

class LogInActivity : AppCompatActivity() {

    private var attemptCounter = 0
    private lateinit var binding: ActivityLogInBinding

    private var userName: String = ""
    private var password: String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLogInBinding.inflate(layoutInflater)
        setContentView(binding.root)
        dataFromPreviousScreen()
        setOnClickListener()
    }

    private fun setOnClickListener() {
        binding.loginId.setOnClickListener {
            attemptCounter++
            val loginUserName = binding.usrNameId.text.toString()
            val loginPassword = binding.pwdId.text.toString()
            if (attemptLogin(loginUserName, loginPassword)) {
                //Navigate to success activity
                val intent = Intent(this@LogInActivity, SuccessActivity::class.java)
                startActivity(intent)
            } else {
                if (attemptCounter >= 2) {
                    // Disable the Login Button
                    binding.loginId.isEnabled = false
                }
            }
        }
    }

    private fun attemptLogin(loginUserName: String, loginPassword: String): Boolean {
        return if (userName == loginUserName && password == loginPassword) {
            true
        } else {
            Toast.makeText(this@LogInActivity, "Login Failed", Toast.LENGTH_SHORT).show()
            false
        }
    }

    private fun dataFromPreviousScreen() {
        val bundle = intent.extras
        if (bundle != null) {
            userName = bundle.getString("userName").toString()
            password = bundle.getString("password").toString()
        }
     }

}